agent/engine/serviceconnect/manager_linux.go

```diff
@@ -200,6 +202,37 @@ func defaultMkdirAllAndChown(path string, perm fs.FileMode, uid, gid int) error
 	return nil
 }
 
+func getRegionFromContainerInstanceARN(containerInstanceARN string) string {
+	// Parse the ARN
+	parsedARN, err := arn.Parse(containerInstanceARN)
+	if err != nil {
+		return ""
+	}
+
+	// Extract the region from the parsed ARN
+	return parsedARN.Region
+}
+
+func isIsoRegion(region string) bool {
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !ok {
```

Comment:
I don't believe AWS SDK Go v1 (which is being used here) is guaranteed to receive updates for newer AWS regions. If that is the case, that means newer regions will default to being marked as an ISO region here, regardless of whether they are ISO or not. Is there some other more specific criteria we can use to determine if a region is ISO?

Reply:
Discussed offline, see https://github.com/aws/amazon-ecs-agent/pull/4448#discussion_r1872342602 for summary of discussion.